This patch was developed in-house and is planned to be contributed upstream.
It is still needed for git 2.14.2.

--- git-2.14.2/git-compat-util.h.orig	2017-10-06 12:18:48.129836956 -0700
+++ git-2.14.2/git-compat-util.h	2017-10-06 12:19:33.797736343 -0700
@@ -741,7 +741,7 @@
 #endif
 
 #ifdef NO_INET_NTOP
-const char *inet_ntop(int af, const void *src, char *dst, size_t size);
+const char *inet_ntop(int af, const void *src, char *dst, socklen_t size);
 #endif
 
 #ifdef NO_PTHREADS
--- git-2.14.2/compat/inet_ntop.c.orig	2017-10-06 12:22:24.935633562 -0700
+++ git-2.14.2/compat/inet_ntop.c	2017-10-06 12:23:37.976662604 -0700
@@ -168,7 +168,7 @@
  *	Paul Vixie, 1996.
  */
 const char *
-inet_ntop(int af, const void *src, char *dst, size_t size)
+inet_ntop(int af, const void *src, char *dst, socklen_t size)
 {
 	switch (af) {
 	case AF_INET:
